fickling/fickle.py

```diff
@@ -1123,6 +1123,26 @@ def run(self, interpreter: Interpreter):
         return objs
 
 
+class Obj(Opcode):
+    name = "OBJ"
+
+    def run(self, interpreter: Interpreter):
+        args = []
+        while interpreter.stack:
+            arg = interpreter.stack.pop()
+            if isinstance(arg, MarkObject):
+                break
+            args.append(arg)
+        else:
+            raise ValueError("Exhausted the stack while searching for a MarkObject!")
+        kls = args.pop()
+        # TODO Verify paths for correctness
+        if args or hasattr(kls, "__getinitargs__") or not isinstance(kls, type):
+            interpreter.stack.append(ast.Call(kls, [list(args)], []))
+        else:
+            interpreter.stack.append(ast.Call(kls, kls, []))
```

Comment:
The arguments will be in the wrong order here. They are appended to the list from last to first, so we should invert the list order to recover the actual order of arguments.

```suggestion
    def run(self, interpreter: Interpreter):
        args = []
        while interpreter.stack:
            arg = interpreter.stack.pop()
            if isinstance(arg, MarkObject):
                break
            args.insert(0, arg)
        else:
            raise ValueError("Exhausted the stack while searching for a MarkObject!")
        kls = args.pop(0)
        # TODO Verify paths for correctness
        if args or hasattr(kls, "__getinitargs__") or not isinstance(kls, type):
            interpreter.stack.append(ast.Call(kls, args, []))
        else:
            interpreter.stack.append(ast.Call(kls, kls, []))
```